core/base/src/TROOT.cxx

```diff
@@ -2574,14 +2544,258 @@ const char**& TROOT::GetExtraInterpreterArgs() {
 }
 
 ////////////////////////////////////////////////////////////////////////////////
+
+#ifdef ROOTPREFIX
+static Bool_t IgnorePrefix() {
+   static Bool_t ignorePrefix = kFALSE;
+   static Bool_t init = kFALSE;
+   if (!init) {
+      if (gSystem->Getenv("ROOTIGNOREPREFIX"))
+         ignorePrefix = kTRUE;
+      init = kTRUE;
+   }
+   return ignorePrefix;
+}
+#endif
+
+////////////////////////////////////////////////////////////////////////////////
+/// Get the rootsys directory in the installation. Static utility function.
+
+const TString& TROOT::GetRootSys() {
+#ifdef ROOTPREFIX
+   if (IgnorePrefix()) {
+#endif
+      static TString rootsys;
+      if (rootsys.IsNull())
```

Comment:
I'm scared about the thread-safety of this... Can we do
```
const char* GetRootSysImpl() {
  if (const char* rsEnv = gSystem->Getenv("ROOTSYS")) {
    return rsEnv;
  }
  return gRootDir;
}
```
and then
```
static TString rootsys = GetRootSysImpl();
```
(or something that actually compiles, and enforces thread safety through initialization?)